src/app/layout/loading/loading.component.ts

```diff
@@ -46,16 +48,21 @@ export class LoadingComponent implements OnInit, AfterViewInit {
 
     // Subscribe to isLoading event
     this.loadingService.isLoading.subscribe((isLoading: number) => {
-      let $setTimeout = setTimeout(() => {
+      const $setTimeout = setTimeout(() => {
         if (isLoading > 0) {
           // Show if isLoading has requests
           this.showLoadingOverlay = true;
           this.spinner.show();
-        }
-        else {
+        } else {
           // Hide if isLoading has lesser requests
           this.showLoadingOverlay = false;
-          this.spinner.hide();
+          of(null)
+              .pipe(
+                  take(1),
+                  delay(40))
+              .subscribe(() => {
+                this.spinner.hide();
+              });
```

Comment:
May I know why we need the delay here?

Reply:
this is one of the ways to take actions out of the stream
same as setTimeout